nixos/modules/services/misc/sssd.nix

```diff
@@ -16,7 +17,7 @@ in {
     };
 
     nixpkgs.config.packageOverrides = pkgs: {
-      sssd = pkgs.sssd.override { nscdConfig = nscd.confFile; };
+      sssd = pkgs.sssd.override { nscdConfig = nscdConfig; };
```

Comment:
This will keep the problem because `source` points to the changing path. Instead I'd drop `nscdConfig` argument and just hardcode it to `/etc/nscd.conf` in the package.. This way you'd also get rid of an override.